### PR TITLE
Create a new package for generic commands like get

### PR DESCRIPTION
**Pull request by smarterclayton**
This is in preparation for cleaning up the structure of `kubectl get` to ease implementing serverside get and streaming API responses. Moves a hardcoded constant into a method in preparation for a future serverside implementation. Also improves some existing description text.

### CODE REVIEW

**Comment by smarterclayton** [comment]
@kubernetes/sig-cli-maintainers

**Comment by smarterclayton** [comment]
@fabianofranz

File: pkg/kubectl/cmd/explain.go [hunk 1]
```diff
@@ -32,9 +32,15 @@ import (
...
-
-		` + validResources)
+		List the fields for supported resources
+		
+		This command describes the fields associated with each supported API resource.
+		Fields are identified via a simple JSONPath identifier: 
+		
+			<type>.<fieldName>[.<fieldName>]
+			
+		Add the --recursive flag to display all of the fields at once without descriptions
```
Comment (fabianofranz) [inline pkg/kubectl/cmd/explain.go]: Missing `.` at the end of this line.

**Comment by fabianofranz** [comment]
One nit then LGTM.

**Comment by smarterclayton** [comment]
de-nitted

**Comment by smarterclayton** [comment]
Rebased, applied labels

**Comment by smarterclayton** [comment]
/retest
Does not need rebase

**Comment by smarterclayton** [comment]
Wat

**Pull request merged**